src/Common/isLocalAddress.cpp

```diff
@@ -81,10 +81,41 @@ struct NetworkInterfaces
 
 bool isLocalAddress(const Poco::Net::IPAddress & address)
 {
+    /** 127.0.0.1 is treat as local address unconditionally.
+      * ::1 is also treat as local address unconditionally.
+      *
+      * 127.0.0.{2..255} are not treat as local addresses, because they are used in tests
+      *  to emulate distributed queries across localhost.
+      *
+      * But 127.{0,1}.{0,1}.{0,1} are treat as local addresses,
+      *  because they are used in Debian for localhost.
+      */
+    if (address.isLoopback())
```

Comment:
Hi @alesapin,
I am afraid this will work only for IP address 127.0.0.1, related to https://pocoproject.org/docs/Poco.Net.IPAddress.html#24435 where documentation says loopback address is 127.0.0.1, otherwise it will never reach the internal part of if statement. I have this trouble in docker environment and I have 127.0.1.1 in /etc/hosts and unfortunately is_local is still 0.

Reply:
Poco source
https://github.com/ClickHouse-Extras/poco/blob/5994506908028612869fee627d68d8212dfe7c1e/Net/src/IPAddressImpl.cpp#L182

So rather poco docs are not in sync, and not clear what is happening in your case

Reply:
BTW - are you sure you use 21.7?

Reply:
Hi @filimonov, sorry i didn't catch the difference between code and docs in poco. Unfortunately no, I am using 21.6.4 - I have clickhouse running in docker and there is no 21.7 docker image at the moment. Once there is 21.7 I will check if it works. 

At the moment I solved it by setting real IP address of the server in /etc/hosts instead of 127.0.1.1 and it works as expected.